### PR TITLE
Add new test "subscribe newsletter" 

### PR DESCRIPTION
**Pull request by SD1982**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Add the newsletter subscription test
| Type?             | refacto 
| Category?         | TE
| BC breaks?        | no
| Deprecations?     |no
| Fixed ticket?     | Fixes #{issue number here}.
| How to test?      | can be tested with this command `TEST_PATH="functional/FO/04_newsletter/01_subscribeNewsletter" npm run specific-test`
| Possible impacts? | NO

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24024)

### CODE REVIEW

File: tests/UI/pages/FO/myAccount/identity.js [hunk 1]
```diff
@@ -56,6 +57,20 @@ class AccountIdentity extends FOBasePage {
 
     return this.getTextContent(page, this.alertSuccessBlock);
   }
+
+  async unsubscribeNewsletter(page, password) {
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/myAccount/identity.js]: Please add description

File: tests/UI/pages/FO/home/index.js [hunk 2]
```diff
@@ -320,6 +323,19 @@ class Home extends FOBasePage {
 
     return this.openLinkWithTargetBlank(page, selector, 'body');
   }
+
+  /**
+   *
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/home/index.js:337]: Please add description

File: tests/UI/pages/BO/modules/psEmailSubscription/index.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ class PsEmailSubscription extends ModuleConfiguration.constructor {
     this.newsletterTableRow = row => `${this.newsletterTableRows}:nth-child(${row})`;
     this.newsletterTableEmptyColumn = `${this.newsletterTableRows} td.list-empty`;
     this.newsletterTableEmailColumn = row => `${this.newsletterTableRow(row)} td:nth-child(5)`;
+
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/modules/psEmailSubscription/index.js]:
```suggestion
```

This new line looks useless, not increasing readability

File: tests/UI/pages/FO/home/index.js [hunk 4]
```diff
@@ -54,6 +58,10 @@ class Home extends FOBasePage {
     this.cartModalSubtotalBlock = `${this.cartContentBlock} .subtotal.value`;
     this.cartModalproductTaxInclBlock = `${this.cartContentBlock} .product-total .value`;
     this.cartModalCheckoutLink = `${this.blockCartModalDiv} div.cart-content-btn a`;
+
+    //Newsletter subscription messages
```
Comment (NeOMakinG) [inline tests/UI/pages/FO/home/index.js]:
```suggestion
    // Newsletter subscription messages
```

File: tests/UI/pages/FO/home/index.js [hunk 5]
```diff
@@ -320,6 +328,19 @@ class Home extends FOBasePage {
 
     return this.openLinkWithTargetBlank(page, selector, 'body');
   }
+
+  /**
+   * subscribe to the newsletter from FO homepage
+   * @param page
+   * @param email
```
Comment (NeOMakinG) [inline tests/UI/pages/FO/home/index.js]:
```suggestion
   * @param {object} page
   * @param {string} email
   *
```

At least, or maybe page has a type?

File: tests/UI/pages/FO/myAccount/identity.js [hunk 6]
```diff
@@ -56,6 +57,26 @@ class AccountIdentity extends FOBasePage {
 
     return this.getTextContent(page, this.alertSuccessBlock);
   }
+
+  /**
+   * Unsubscribe from the newletter from customer edit information page
+   * @param page
+   * @param password
+   * @returns {Promise<string|TextContent|*>}
```
Comment (NeOMakinG) [inline tests/UI/pages/FO/myAccount/identity.js]:
```suggestion
   * @param {object} page
   * @param {string} password
   *
   * @returns {Promise<string|TextContent|*>}
```

**Comment by NeOMakinG** [comment]
Some simple code style feedbacks, overwise it looks fine

File: tests/UI/pages/FO/myAccount/identity.js [hunk 7]
```diff
@@ -19,6 +19,7 @@ class AccountIdentity extends FOBasePage {
     this.birthdateInput = `${this.createAccountForm} input[name='birthday']`;
     this.customerPrivacyCheckbox = `${this.createAccountForm} input[name='customer_privacy']`;
     this.psgdprCheckbox = `${this.createAccountForm} input[name='psgdpr']`;
+    this.newsletterCheckbox = `${this.createAccountForm} input[name='newsletter']`;
```
Comment (Progi1984) [inline tests/UI/pages/FO/myAccount/identity.js]:
```suggestion
    this.newsletterCheckbox = `${this.createAccountForm} input[name=newsletter]`;
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback

File: tests/UI/pages/FO/home/index.js [hunk 8]
```diff
@@ -320,6 +328,20 @@ class Home extends FOBasePage {
 
     return this.openLinkWithTargetBlank(page, selector, 'body');
   }
+
+  /**
+   * subscribe to the newsletter from FO homepage
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/home/index.js]:
```suggestion
   * Subscribe to the newsletter from the FO homepage
```

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by boubkerbribri — DISMISSED** [review]
LGTM

**Comment by NeOMakinG** [comment]
I think there is a problem with your base branch, too much commits are unrelated with your work

**Comment by boubkerbribri** [comment]
> I think there is a problem with your base branch, too much commits are unrelated with your work

It should be good now.

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @SD1982

**Pull request merged**